spec/integ/crypto/verification.spec.ts

```diff
@@ -990,6 +992,37 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("verification (%s)", (backend: st
             aliceClient.setGlobalErrorOnUnknownDevices(false);
             syncResponder.sendOrQueueSyncResponse(getSyncResponse([BOB_TEST_USER_ID]));
             await syncPromise(aliceClient);
+
+            // Rust crypto requires the sender's device keys before it accepts a
+            // verification request.
+            if (backend === "rust-sdk") {
+                // Cast to a RustCrypto object, since we call some
+                // RustCrypto-specific functions
+                const crypto = aliceClient.getCrypto()!;
+                expect(crypto).toBeInstanceOf(RustCrypto);
+                const rustCrypto = crypto as RustCrypto;
+
+                const bobIdentityKeys = JSON.parse(testOlmAccount.identity_keys());
+                const bobDeviceKeys: IDeviceKeys = {
+                    user_id: "@bob:xyz",
+                    device_id: "BobDevice",
+                    algorithms: ["m.olm.v1.curve25519-aes-sha2", "m.megolm.v1.aes-sha2"],
+                    keys: {
+                        "curve25519:BobDevice": bobIdentityKeys.curve25519,
+                        "ed25519:BobDevice": bobIdentityKeys.ed25519,
+                    },
+                };
+                const signature = testOlmAccount.sign(anotherjson.stringify(bobDeviceKeys));
+                bobDeviceKeys.signatures = {
+                    "@bob:xyz": {
+                        "ed25519:BobDevice": signature,
+                    },
+                };
+                e2eKeyResponder.addDeviceKeys(bobDeviceKeys);
+                await rustCrypto.processDeviceLists({ changed: ["@bob:xyz"] });
+                rustCrypto.onSyncCompleted({});
+                await rustCrypto.getUserDeviceInfo(["@bob:xyz"]);
```

Comment:
poking things directly into the private methods of the crypto stack feels like a violation of this being an "integration" test -- the premise of such tests is that they operate via the *public* api of the js-sdk.

why can't we return this information via the `syncResponder` ?

Reply:
I had tried various things to get RustCrypto to get it to see the device, including using `syncResponder`.  The problem was that I couldn't figure out the right combination of things to `await` so that it would reliably get the device before it tried to verify.  I don't remember exactly which combinations I tried, but I tried a few different things, before concluding that it looked like I would have to hit *some* RustCrypto method.

Reply:
That seems to work. I don't remember exactly what I had tried before, so I don't know why what I tried didn't work.